### PR TITLE
Get precise data

### PR DESCRIPTION
**Pull request by matburnx**
Added `rawPercentage` to have the percentages present in the file as strings and keep the precision of it.
Added tests to it.
Now the method `hardestCall` uses this `rawPercentage` to decide which trace took the most time.

### CODE REVIEW

File: src/PerfTreeParser-Tests/PerfTreeParserTest.class.st [hunk 1]
```diff
@@ -468,6 +468,41 @@ PerfTreeParserTest >> testPercentageOf [
...
+		equals: (testParserMultipleChildren rawPercentageOf: 8).
+
+	"Case with same percentage"
+	self
+		assert: (testParserSamePercentage rawPercentageOf: 4)
+		equals: (testParserSamePercentage rawPercentageOf: 5).
+
+	self
+		assert: (testParserSamePercentage rawPercentageOf: 5)
+		equals: (testParserSamePercentage rawPercentageOf: 6)
```
Comment (guillep) [inline src/PerfTreeParser-Tests/PerfTreeParserTest.class.st:503]: As a general rule, we should try to not put many unrelated asserts together in the same test.
They are testing different cases, so this means that if the first fails we will not have information about the ones below.

Ideally, they should be split in many tests

File: src/PerfTreeParser/PerfTreeParser.class.st [hunk 2]
```diff
@@ -165,25 +166,39 @@ PerfTreeParser >> numberOfSpacesAt: aNodeIndex [
...
+	"In the case of multiples functions for one percentage, it gives the same percentage to every functions."
+
+	| stringLine |
+	(aLine between: 1 and: self numberOfFunctions) ifFalse: [ ^ '' ].
+
+	stringLine := ((((self readLine: aLine) copyUpTo: $%) last: 5)
+		               copyWithout: $-) copyWithout: Character space.
+
+	stringLine = ((self readLine: aLine) last: stringLine size) ifTrue: [
+		^ self rawPercentageOf: aLine - 1 ].
```
Comment (guillep) [inline src/PerfTreeParser/PerfTreeParser.class.st:197]: Why do we need to re-parse?
The parser could remember the last percentage it saw, right?

**Review by guillep — CHANGES_REQUESTED** [review]
Thanks @matburnx I did some comments

File: src/PerfTreeParser/PerfTreeParser.class.st (hunk 2 above)
Comment (matburnx) [reply src/PerfTreeParser/PerfTreeParser.class.st:197]: Unfortunately, as the parsing is done line by line. I don't think it is possible not to parse the line above again. 

However, it happens only if the line has no percentage, then it gets the one above until a line has a percentage written to it.

**Comment by matburnx** [comment]
>     * Commit: [b72cb35](https://github.com/Alamvic/XCVMProfiler/commit/b72cb35a774ba66f42f5c29e9dd6523a7e1bd890)
> 
>     * Pull Request: nil
It's a bit weird that it doesn't show the pull request ID.

File: src/PerfTreeParser-Tests/PerfTreeParserTest.class.st (hunk 1 above)
Comment (matburnx) [reply src/PerfTreeParser-Tests/PerfTreeParserTest.class.st]: I split the tests for every big test method that had the same issue

File: src/PerfTreeParser/PerfTreeParser.class.st (hunk 2 above)
Comment (guillep) [reply src/PerfTreeParser/PerfTreeParser.class.st:197]: I mean, the parser can remember the percentage of the previous line!
Actually, it's already parsed and stored in the node :)

**Comment by guillep** [comment]
> > * Commit: [b72cb35](https://github.com/Alamvic/XCVMProfiler/commit/b72cb35a774ba66f42f5c29e9dd6523a7e1bd890)
> > * Pull Request: nil
> 
> It's a bit weird that it doesn't show the pull request ID.

Haha yes, I don't know, to be fixed.

**Review by guillep — APPROVED** [review]
Super good job! Thanks @matburnx !

File: src/PerfTreeParser/PerfTreeParser.class.st (hunk 2 above)
Comment (matburnx) [reply src/PerfTreeParser/PerfTreeParser.class.st:197]: I'll try to do something about this

**Pull request merged**